### PR TITLE
add .editorconfig file

### PR DESCRIPTION
**Pull request by TysonMN**
Partially addresses #271

[EditorConfig](https://editorconfig.org/) is a IDE-independent way of expressing some formatting settings.

The file contains my preferred settings.  Feel free to suggest different settings.  I care much more that a setting exists than whether it is set to my preference.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+root = true
+
+[*] # rules that follow apply to files with any extension
+insert_final_newline = false
```
Comment (moodmosaic) [inline .editorconfig:4]: I think we're inserting a final new line everywhere, or in most places, but I could be wrong.

(I'm out and about at the moment so I can't check.)

**Review by moodmosaic — COMMENTED** [review]
👍

File: .editorconfig (hunk 1 above)
Comment (moodmosaic) [reply .editorconfig:4]: If that's the case then this must be set to true.

Comment (TysonMN) [reply .editorconfig]: Yep, I just checked.  I changed that setting.

Comment (moodmosaic) [reply .editorconfig]: 👍

**Comment by moodmosaic** [comment]
💯

**Pull request merged**